ebpf/aya-ebpf/src/programs/sk_buff.rs

```diff
@@ -90,6 +90,10 @@ impl SkBuff {
         let len = usize::try_from(self.len()).map_err(|core::num::TryFromIntError { .. }| -1)?;
         let len = len.checked_sub(offset).ok_or(-1)?;
         let len = len.min(dst.len());
+        let in_bounds = check_bounds_signed(len as c_long, 0, dst.len() as c_long + 1);
```

Comment:
This can be done without a let binding.

```suggestion
        if !check_bounds_signed(len as c_long, 0, dst.len() as c_long + 1) {
            return Err(-1);
        }
```

Also why `dst.len() as c_long + 1` - surely that's out of bounds?

Reply:
`check_bounds_signed` is inclusive on both sides, so this code is wrong. Pushing a new PR with the correct fix and an integration test soon.